nativelib/src/main/scala-3/scala/scalanative/safe/SafeZone.scala

```diff
@@ -0,0 +1,59 @@
+package scala.scalanative.safe
+
+import language.experimental.captureChecking
+import scalanative.unsigned._
+import scala.annotation.implicitNotFound
+import scala.scalanative.runtime.{RawPtr, CMemoryPool, CMemoryPoolZone}
+
+@implicitNotFound("Given method requires an implicit zone.")
+trait SafeZone {
+
+  /** Frees allocations. This zone allocator is not reusable once closed. */
+  def close(): Unit
+
+  /** Return this zone allocator is open or not. */
+  def isOpen: Boolean
+
+  /** Return this zone allocator is closed or not. */
+  def isClosed: Boolean = !isOpen
+
+  /** Return the handle of this zone allocator. */
+  def handle: RawPtr
```

Comment:
I feel like this shouldn't be user accessible without some explicit "unsafe" signal, e.g. perhaps
```scala
scala.scalanative.unsafe.handleOfSafeZone(sz)
```

Reply:
or alternatively instead of `RawPtr`, maybe an opaque type alias is used?

Reply:
I think that making it `private[scalanative] def handle: RawPtr` should do the job. In general `RawPtr` by definition is `super-unsafe` and it should be used only by the internals. `RawPtr` can wrapped by `Ptr[T]` which is just `unsafe` ;)